dev/dev-env-setup.sh

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env bash
+
+set +exv
+
+showHelp() {
+cat << EOF
+Usage: ./install-dev-env.sh [-d] [directory to install virtual environment] [-v] [-q] [-f]
+Development environment setup script for Python in linux-based Operationg Systems (including OSX).
+Note: this script will not work on Windows.
+This script will:
+
+  - Install pyenv if not installed
+  - Retrieve the appropriate Python version (minimum required) for compatibility support
+  - Check if the virtual environment already exists
+    - If it does, prompt for replacement
+      - If replacing, delete old virtual environment.
+  - Create a virtual environment using the minimum required Python version based on previous step logic
+  - Activate the environment
+  - Install required dependencies for the dev envrionment
+
+  Example usage:
+
+  From root of MLflow repository on local with a destination virtualenv path of <MLFLOW_HOME>/.venvs/mlflow-dev:
```

Comment:
```suggestion
  From root of MLflow repository on local with a destination virtualenv path of <MLFLOW_HOME>/.venv:
```

Can we use `.venv` instead of `.venvs/mlflow-dev-env`?

Reply:
I saw https://github.com/mlflow/mlflow/pull/5717/files#diff-76e7ca2f2843ac5f659b45382768c1b64bf8704e8d3c6189c4e3623f5d465bc2R162-R170. Now the current approach makes sense.